components/support/jwcrypto/src/direct.rs

```diff
@@ -0,0 +1,41 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::{
+    aes_gcm_decrypt, aes_gcm_encrypt,
+    error::{JwCryptoError, Result},
+    Algorithm, CompactJwe, EncryptionAlgorithm, JweHeader, Jwk, JwkKeyParameters,
+};
+
+pub(crate) fn encrypt_to_jwe(data: &[u8], jwk: &Jwk) -> Result<CompactJwe> {
+    let protected_header = JweHeader {
+        kid: jwk.kid.clone(),
+        alg: Algorithm::Direct,
+        enc: EncryptionAlgorithm::A256GCM, // ??
+        epk: None,
+        apu: None,
+        apv: None,
+    };
+    let secret = match jwk.key_parameters {
+        JwkKeyParameters::Direct(ref bytes) => bytes,
+        _ => return Err(JwCryptoError::IllegalState("Not a Direct key")),
+    };
+    aes_gcm_encrypt(data, protected_header, &secret)
+}
+
+pub(crate) fn decrypt_jwe(jwe: &CompactJwe, jwk: Jwk) -> Result<String> {
+    let protected_header = JweHeader {
+        kid: jwk.kid.clone(),
+        alg: Algorithm::Direct,
+        enc: EncryptionAlgorithm::A256GCM, // ??
+        epk: None,
+        apu: None,
+        apv: None,
+    };
```

Comment:
I'm surprised that you need to create a `protected_header` here, shouldn't the header be part of the jwe that you are trying to decrypt? This might be some factoring weirdness from the ECDH implementation that we need to clean up.

Reply:
Looking at the ECDH version, it seems to get the protected header via `jwe.protected_header()?` rather than constructing a new one.